File: contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums: list[int]) -> bool:
        seen = set()
        for num in nums:
            if num in seen:   # already seen → duplicate found
                return True
            seen.add(num)
        return False


# Time: O(n)
# Space: O(n)

File: contains_duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        for (int i = 0; i < nums.size(); i++) {
            for (int j = i + 1; j < nums.size(); j++) {
                if (nums[i] == nums[j]) {
                    return true;
                }
            }
        }
        return false;
    }
};


/* 
Time Complexity:  O(N*N), Because we are traversing the whole array again and again for every integer.
Space Complexity: O(1), Since, we are not using any extra space. */
